### ISSUE TITLE
Segfault when trying to load rss from Tiny Tiny RSS with empty ttrss-password

### ISSUE THREAD

**Issue by serpent7776** [akrennmair/newsbeuter#336]
I'm experiencing strange segfault on app start. I'm loading rss from Tiny Tiny RSS with configuration:

    urls-source "ttrss"

I'm not specifying `ttrss-password` value in config file, only `ttrss-url` and `ttrss-login`.

I did two things recently that might affect this:
- I updated Tiny Tiny RSS
- I updated my FreeBSD system on which I run newsbeuter

but I suspect that Tiny Tiny RSS update is more relevant here.

I got this when starting app:

    XDG: configuration directory '/home/serpent/.config/newsbeuter' not accessible, using '/home/serpent/.newsbeuter' instead.
    Starting newsbeuter 2.9...
    Loading configuration...done.
    Opening cache...done.
    Loading URLs from Tiny Tiny RSS...Segmentation fault

gdb gives this info:

```
0x000000000047dcb1 in newsbeuter::ttrss_api::retrieve_sid (this=0x8058da1a0) at src/ttrss_api.cpp:60
60      src/ttrss_api.cpp: No such file or directory.
        in src/ttrss_api.cpp
```

This references [this](https://github.com/akrennmair/newsbeuter/blob/newsbeuter-2.9/src/ttrss_api.cpp#L60) file I guess

When I change config to include `ttrss-password` everything works as expected.
Any ideas what is happening here?

**Comment by Minoru** [comment]
Thanks for reporting! Can you build from `bugfix/336` branch and check if that fixes the issue?

**Comment by serpent7776** [comment]
I can confirm `bugfix/336` branch fixes the issue, thanks.

**Comment by Minoru** [comment]
Thanks for checking! The fix will be released with version 2.10.
